webdriver/tests/bidi/browsing_context/capture_screenshot/capture_screenshot.py

```diff
@@ -35,3 +36,25 @@ async def test_capture(bidi_session, top_context, inline, compare_png_bidi):
 
     comparison = await compare_png_bidi(new_data, data)
     assert comparison.equal()
+
+@pytest.mark.asyncio
+async def test_capture_with_viewport(bidi_session, new_tab, inline, compare_png_bidi):
+    dpr = await get_device_pixel_ratio(bidi_session, new_tab)
+
+    test_viewport = {"width": 499, "height": 599}
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        viewport=test_viewport)
+
+    expected_size = {
+        "width": test_viewport["width"] * dpr,
+        "height": test_viewport["height"] * dpr
```

Comment:
Please use `floor` to cut of the fragments from the result given that dpr is a float value.

Reply:
dpr should not be a float (see get_device_pixel_ratio helper where it is floored)

Reply:
So that was most likely an oversight and we shouldn't have floored it there. In case of a DPR of `1.5` and a viewport width of 100 we current would get `100` pixels, while it should be `150` pixels. Maybe we should better improve `get_viewport_dimensions` by adding a eg. `useDevicePixelRatio` flag to it which directly computes it right. What do you think? IMHO it should be fine in a follow-up.

Reply:
Updated in this PR. I wonder if browser implementations would floor or ceil when taking a screenshot 🤔